alg/teca_temporal_reduction.cxx

```diff
@@ -0,0 +1,1739 @@
+            req_arrays.insert(vv_mask);
+
+            // get the fill value
+            teca_metadata array_atts;
+            atts.get(array, array_atts);
+            if (this->fill_value != -1)
+            {
+                fill_value = this->fill_value;
+            }
+            else if (array_atts.has("_FillValue"))
+            {
+                array_atts.get("_FillValue", fill_value);
+            }
+            else if (array_atts.has("missing_value"))
+            {
+                array_atts.get("missing_value", fill_value);
+            }
+        }
+
+        // create and initialize the operator
+        p_reduction_operator op
+                      = reduction_operator_factory::New(this->operator_name);
+        op->initialize(fill_value);
+
+        // save the operator
+        this->op[array] = op;
+    }
+
+    // generate one request for each time step in the interval
+    std::vector<teca_metadata> up_reqs;
+    std::string request_key;
+    md.get("index_request_key", request_key);
+    unsigned long req_id;
+    req_in.get(request_key, req_id);
+    int i = this->indices[req_id].start_index;
+    while (i <= this->indices[req_id].end_index)
+    {
+        teca_metadata req(req_in);
+        req.set("arrays", req_arrays);
+        req.set(request_key, i);
+        up_reqs.push_back(req);
+        i += 1;
+    }
+
+    return up_reqs;
+}
+
+// --------------------------------------------------------------------------
+const_p_teca_dataset teca_temporal_reduction::execute(
+    unsigned int port,
+    const std::vector<const_p_teca_dataset> &data_in,
+    const teca_metadata &req_in,
+    int streaming)
+{
+    (void)port;
+
+    // get the requested ind
+    std::string request_key;
+    req_in.get("index_request_key", request_key);
+    unsigned long req_id;
+    req_in.get(request_key, req_id);
+
+    int device_id = -1;
+    req_in.get("device_id", device_id);
+
+#ifdef TECA_DEBUG
+    std::cerr << teca_parallel_id()
+        << "teca_temporal_reduction::execute request "
+        << req_id << " device " << device_id
+        << " (" << this->indices[req_id].start_index
+        << " - " << this->indices[req_id].end_index
+        << "), reducing " << data_in.size() << ", "
+        << streaming << " remain" << std::endl;
+#endif
+
+    size_t n_data = data_in.size();
+
+    // copy the first mesh
+    const_p_teca_cartesian_mesh mesh_in
+      = std::dynamic_pointer_cast<const teca_cartesian_mesh>(data_in[n_data-1]);
+    p_teca_cartesian_mesh mesh_out = teca_cartesian_mesh::New();
+    mesh_out->shallow_copy(
+      std::const_pointer_cast<teca_cartesian_mesh>(mesh_in));
+    p_teca_array_collection arrays_out = mesh_out->get_point_arrays();
+
+
+    // accumulate incoming values
+    for (int i = n_data-2; i >= 0; --i)
+    {
+        mesh_in
+           = std::dynamic_pointer_cast<const teca_cartesian_mesh>(data_in[i]);
+        const_p_teca_array_collection arrays_in = mesh_in->get_point_arrays();
```

Comment:
I find it counter intuitive as to why we are processing the list in reverse order.  I suspect it is a hold over of the Python implementation which pop'd elements off the list. I don't think the order matters here. It's probably best to process in ascending order, just because thats what we always do. If not, comment the heck out of it.

Also, pull calculations of the loop termination condition.